src/common/set.c

```diff
@@ -3796,12 +3796,17 @@ util_pool_open_nocheck(struct pool_set *set, unsigned flags)
 
 	if (flags & POOL_OPEN_CHECK_BAD_BLOCKS) {
 		/* check if any bad block recovery file exists */
-		if (badblocks_recovery_file_exists(set)) {
+		int bfe = badblocks_recovery_file_exists(set);
+		if (bfe > 0) {
 			ERR(
 				"error: a bad block recovery file exists, run 'pmempool sync --bad-blocks' utility to try to recover the pool");
 			errno = EINVAL;
 			return -1;
 		}
+		if (bfe < 0) {
+			LOG(1, "there is no bad block recovery file");
```

Comment:
that's not true, a negative value means there was an error during checking whether recovery file exists